src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -4486,6 +4659,68 @@ SendColocationMetadataCommands(MetadataSyncContext *context)
 }
 
 
+/*
+ * SendTenantSchemaMetadataCommands sends tenant schema metadata entries with
+ * transactional or nontransactional mode according to transactionMode inside
+ * metadataSyncContext.
+ */
+void
+SendTenantSchemaMetadataCommands(MetadataSyncContext *context)
+{
+	ScanKeyData scanKey[1];
+	int scanKeyCount = 0;
+
+	Relation pgDistTenantSchema = table_open(DistTenantSchemaRelationId(),
+											 AccessShareLock);
+	SysScanDesc scanDesc = systable_beginscan(pgDistTenantSchema, InvalidOid, false, NULL,
+											  scanKeyCount, scanKey);
+
+	MemoryContext oldContext = MemoryContextSwitchTo(context->context);
+	HeapTuple heapTuple = NULL;
+	while (true)
+	{
+		ResetMetadataSyncMemoryContext(context);
+
+		heapTuple = systable_getnext(scanDesc);
+		if (!HeapTupleIsValid(heapTuple))
+		{
+			break;
+		}
+
+		Form_pg_dist_tenant_schema tenantSchemaForm =
+			(Form_pg_dist_tenant_schema) GETSTRUCT(heapTuple);
+
+		/*
+		 * Not directly read colocation id from FormData because it might
+		 * be set to NULL.
+		 */
+		bool colocationIdIsNull = false;
+		uint32 colocationId = DatumGetUInt32(
+			heap_getattr(heapTuple,
+						 Anum_pg_dist_tenant_schema_colocationid,
+						 RelationGetDescr(pgDistTenantSchema),
+						 &colocationIdIsNull));
+		if (colocationIdIsNull)
+		{
+			colocationId = INVALID_COLOCATION_ID;
+		}
+
+		StringInfo insertTenantSchemaCommand = makeStringInfo();
+		appendStringInfo(insertTenantSchemaCommand,
+						 "SELECT pg_catalog.citus_internal_insert_tenant_schema(%s, %u)",
+						 RemoteSchemaIdExpression(tenantSchemaForm->schemaid),
+						 colocationId);
+
+		List *commandList = list_make1(insertTenantSchemaCommand->data);
+		SendOrCollectCommandListToActivatedNodes(context, commandList);
+	}
+	MemoryContextSwitchTo(oldContext);
```

Comment:
@aykut-bozkurt, I'd like to hear whether I'm using the new metadata-sync APIs correctly, to sync the entries of that new metadata table.

Reply:
Looks right.